api/handlers/service_offering.go

```diff
@@ -79,6 +80,30 @@ func (h *ServiceOffering) get(r *http.Request) (*routing.Response, error) {
 	return routing.NewResponse(http.StatusOK).WithBody(presenter.ForServiceOffering(serviceOffering, h.serverURL, includedResources...)), nil
 }
 
+func (h *ServiceOffering) update(r *http.Request) (*routing.Response, error) {
+	authInfo, _ := authorization.InfoFromContext(r.Context())
+	logger := logr.FromContextOrDiscard(r.Context()).WithName("handlers.service-offering.update")
+
+	serviceOfferingGUID := routing.URLParam(r, "guid")
+
+	var payload payloads.ServiceOfferingUpdate
+	if err := h.requestValidator.DecodeAndValidateJSONPayload(r, &payload); err != nil {
+		return nil, apierrors.LogAndReturn(logger, err, "failed to decode payload")
+	}
+
+	_, err := h.serviceOfferingRepo.GetServiceOffering(r.Context(), authInfo, serviceOfferingGUID)
+	if err != nil {
+		return nil, apierrors.LogAndReturn(logger, apierrors.ForbiddenAsNotFound(err), "Error getting service offering in repository")
```

Comment:
The `handlers\service_offering_test` does not cover the case that whenever the repository returns a forbidden error, it is "masked" to `NotFound` (via the `ForbiddenAsNotFound` utility) because not to leak the knowledge that an object with that ID actually exists.

[Here](https://github.com/cloudfoundry/korifi/blob/9090f8b1d77c3442e0e1f42df2774b831c69dec8/api/handlers/process_test.go#L483-L491) is a similar test form the processes handler